dmgforth

```diff
@@ -55,4 +56,5 @@ depth 0<> [if]
 fix-header-complement
 fix-global-checksum
 output-file dump-rom
+s" output.sym" dump-sym
```

Comment:
This should probably be `<output-file>.sym`, but didn't know how to properly do this.

Reply:
String manipulation is pretty tricky in Forth. You can use `pad`, which just pushes the address of a temporary region of memory useful to do string manipulation.

Doing so, I got this working:

```forth
: append-to { c-addr u addr -- addr' }
  c-addr addr u move
  addr u + ;
  
: append-string ( c-addr1 u1 c-addr2 u2 -- c-addr3 u3 )
  2swap dup >r
  pad append-to

  over >r
  append-to

  drop pad r> r> + ;

( Remove the extension of a filename )
: sans-extname ( c-addr u -- c-addr' u' )
  dup 1+ 1 ?do
    2dup i - + c@
    [char] . = if i - unloop exit then
  loop ;

: symbol-file ( c-addr u -- c-addr' u' )
  sans-extname s" .sym" append-string ;

: output-file ( c-addr u -- c-addr' u' )
  sans-extname s" .gb" append-string ;

```

then you can use it:

```
s" input.fs" symbol-file  type ( prints input.sym )
```

Note that pad is a transient memory. The next pad will invalidate it. So you can't store the resulting string in a variable or anything.

You must use it immediately, like in

`input symbol-file create-file`

Reply:
If you want persistent memory, you can use dynamic memory allocation

https://www.complang.tuwien.ac.at/forth/gforth/Docs-html/Heap-Allocation.html#Heap-Allocation

but then it needs to be free-ed later on.  Or you can just use dictionary allocation if you know it won't be a problem by using up all the memory.